File: app/src/main/java/com/ctyeung/popularmoviestage2/data/utilities/MovieHelper.java
package com.ctyeung.popularmoviestage2.data.utilities;

/**
 * Created by ctyeung on 8/19/17.
 */

public class MovieHelper {
    final public static String BASE_URL = "https://api.themoviedb.org/3/movie/";
    final public static String API_KEY_VALUE = "838bb81a894bc77678913ab8b239cfb3";
    final public static String PARAM_API_KEY = "api_key";

    public static final String SORT_POPULAR = "popular";
    public static final String SORT_TOP_RATED = "top_rated";
    public static final String SORT_FAVORITE = "favorite";
    public static final String KEY_POSTER_PATH = "poster_path";
    public static final String KEY_TITLE = "title";
    public static final String KEY_ORIGINAL_TITLE = "original_title";
    public static final String KEY_PLOT = "overview";
    public static final String KEY_RELEASE_DATE = "release_date";
    public static final String KEY_RUNTIME = "runtime";
    public static final String KEY_VOTE_AVERAGE = "vote_average";
    public static final String KEY_TRAILER = "key";
    public static final String KEY_AUTHOR = "author";
    public static final String KEY_REVIEW_URL = "url";
    public static final String KEY_ID = "id";
    public static final String KEY_NAME = "name";
    public static final String KEY_TYPE = "type";

    public final static String BASE_POSTER_URL = "https://image.tmdb.org/t/p/";
    public final static String PARAM_API_VIDEO = "videos";
    public final static String PARAM_API_REVIEW = "reviews";
    public final static String PARAM_API_V = "v";

    public final static int INDEX_THUMBNAIL = 5;
    public final static int INDEX_DETAIL = 3;

    public final static String BASE_YOUTUBE_URL = "https://www.youtube.com/watch?v=";

    public static String getSizeByIndex(int i)
    {
        switch (i) {
            case 0:
                return "w92/";

            case 1:
                return "w154/";

            default:
            case 2:
                return "w185/";

            case 3:
                return "w342/";

            case 4:
                return "w500/";

            case 5:
                return "w780/";

            case 6:
                return "original/";
        }
    }

    public static boolean isVideo(String str)
    {
        boolean bRet = (str.indexOf("key")>0)?true:false;
        return bRet;
    }
}

File: app/src/main/java/com/ctyeung/popularmoviestage2/data/utilities/MovieHelper.kt
package com.ctyeung.popularmoviestage2.data.utilities

/**
 * Created by ctyeung on 8/19/17.
 */
object MovieHelper {
    const val BASE_URL = "https://api.themoviedb.org/3/movie/"
    const val API_KEY_VALUE = "838bb81a894bc77678913ab8b239cfb3"
    const val PARAM_API_KEY = "api_key"
    const val SORT_POPULAR = "popular"
    const val SORT_TOP_RATED = "top_rated"
    const val SORT_FAVORITE = "favorite"
    const val KEY_POSTER_PATH = "poster_path"
    const val KEY_TITLE = "title"
    const val KEY_ORIGINAL_TITLE = "original_title"
    const val KEY_PLOT = "overview"
    const val KEY_RELEASE_DATE = "release_date"
    const val KEY_RUNTIME = "runtime"
    const val KEY_VOTE_AVERAGE = "vote_average"
    const val KEY_TRAILER = "key"
    const val KEY_AUTHOR = "author"
    const val KEY_REVIEW_URL = "url"
    const val KEY_ID = "id"
    const val KEY_NAME = "name"
    const val KEY_TYPE = "type"
    const val BASE_POSTER_URL = "https://image.tmdb.org/t/p/"
    const val PARAM_API_VIDEO = "videos"
    const val PARAM_API_REVIEW = "reviews"
    const val PARAM_API_V = "v"
    const val INDEX_THUMBNAIL = 5
    const val INDEX_DETAIL = 3
    const val IS_FAVORITE = "isFavorite"
    const val BASE_YOUTUBE_URL = "https://www.youtube.com/watch?v="
    fun getSizeByIndex(i: Int): String {
        return when (i) {
            0 -> "w92/"
            1 -> "w154/"
            2 -> "w185/"
            3 -> "w342/"
            4 -> "w500/"
            5 -> "w780/"
            6 -> "original/"
            else -> "w185/"
        }
    }

    fun isVideo(str: String): Boolean {
        return if (str.indexOf("key") > 0) true else false
    }
}
